templates/single_artist_page.html

```diff
@@ -4,17 +4,19 @@
 <h2>{{ query.name }}</h2>
 <h1> located in: {{ query.location }}
 <img src="{{ query.image }}">
-<div> TODO: query for this artist albums and display them</div>
 <ul>
     <li>
         <h4>Description</h4>
         <ul>
             <li>{{ query.description }}</li>
-            <li> more info at: <a href="{{ query.website }}"</li>
+            <a href="{{ query.website }}"><li> more info</li></a>
         </ul>
         <h4>Albums</h4>
+       
         <ul>
-            <a href="/albums/{{ subquery.id }}"><li>{{ subquery.title }} | {{ subquery.release_date }}</a></li>
+            {% for album in albums %}
```

Comment:
This is fine the way it is, or you can do 
{% for album in query.albums %}

See the comment on the app.route() code